packages/core/src/stepFunctionsWorkflowStudio/WorkflowStudioEditor.ts

```diff
@@ -0,0 +1,180 @@
+ * between the webview and the extension context. This class stores the state of the
+ * local file that is being edited in the webview panel, in the property 'fileStates'. The
+ * 'autoSaveFileStates' property is used to store local changes that are being made in the
+ * webview panel.
+ */
+export class WorkflowStudioEditor {
+    public readonly documentUri: vscode.Uri
+    public webviewPanel: vscode.WebviewPanel
+    protected readonly disposables: vscode.Disposable[] = []
+    protected isPanelDisposed = false
+    private readonly onVisualizationDisposeEmitter = new vscode.EventEmitter<void>()
+    private fileId: string
+    public workSpacePath: string
+    public defaultTemplatePath: string
+    public defaultTemplateName: string
+    // TODO: add fileStates and autoSaveFileStates variables to store the state of the file and handle auto-save
+    private getWebviewContent: () => Promise<string>
+
+    public constructor(
+        textDocument: vscode.TextDocument,
+        webviewPanel: vscode.WebviewPanel,
+        context: vscode.ExtensionContext,
+        fileId: string,
+        getWebviewContent: () => Promise<string>
+    ) {
+        this.getWebviewContent = getWebviewContent
+        this.documentUri = textDocument.uri
+        this.webviewPanel = webviewPanel
+        this.workSpacePath = path.dirname(textDocument.uri.fsPath)
+        this.defaultTemplatePath = textDocument.uri.fsPath
+        this.defaultTemplateName = path.basename(this.defaultTemplatePath)
+        this.fileId = fileId
+
+        telemetry.workflowStudio_opened.record({
+            id: this.fileId,
+        })
+
+        this.setupWebviewPanel(textDocument, context)
+    }
+
+    public get onVisualizationDisposeEvent(): vscode.Event<void> {
+        return this.onVisualizationDisposeEmitter.event
+    }
+    public getPanel(): vscode.WebviewPanel | undefined {
+        if (!this.isPanelDisposed) {
+            return this.webviewPanel
+        }
+    }
+
+    public showPanel(): void {
+        this.getPanel()?.reveal()
+    }
+
+    public async refreshPanel(context: vscode.ExtensionContext) {
+        if (!this.isPanelDisposed) {
+            this.webviewPanel.dispose()
+            const document = await vscode.workspace.openTextDocument(this.documentUri)
+            this.setupWebviewPanel(document, context)
+        }
+    }
+
+    protected getText(textDocument: vscode.TextDocument): string {
+        return textDocument.getText()
+    }
+
+    /**
+     * Sets up the webview panel for Workflow Studio Editor. This includes creating the
+     * panel, setting up the webview content, and handling the communication between the webview
+     * and the extension context.
+     * @param textDocument The text document to be displayed in the webview panel.
+     * @param context The extension context.
+     * @private
+     */
+    private setupWebviewPanel(textDocument: vscode.TextDocument, context: vscode.ExtensionContext) {
```

Comment:
The setupWebviewPanel method wraps the resolution of a promise in another promise. The inner promise is redundant.